### PR TITLE
[WIP] Implementation of Bellman-Ford

### PR DESCRIPTION
**Pull request by codeSG**
### - SRC Directory
- [x] bellman_ford.c
- [x] bellman_ford_driver.cpp
- [x] CMakeLists.txt

### - Include/driver Directory
- [x] bellman_ford_driver.h

### - Sql Directory
- [x] bellman_ford.sql
- [x] CMakeLists.txt

### - pgTap Directory
- [x] bellman_ford-compare-dijkstra.sql
- [x] bellman_ford-innerQuery.sql
- [x] bellman_ford-typesCheck.sql

### - Doc Directory
- [x] doc-pgr_bellman_ford.queries
- [x] pgr_bellman_ford.rst
- [x] CMakeLists.txt

### - Test Directory
- [x] doc-bellman_ford.result
- [x] doc-bellman_ford.test.sql
- [x] test.conf
<br>

- [x] Modified configurations.conf File

### CODE REVIEW

**Comment by cvvergara** [comment]
I think we will live with that un necessarily merge

**Comment by cvvergara** [comment]
Other emtpy commit
https://github.com/pgRouting/pgrouting/pull/1033/commits/61c554a164d05958a2b0763d64c461aed73aff31

File: src/bellman_ford/src/bellman_ford.c [hunk 1]
```diff
@@ -0,0 +1,306 @@
+/*PGR-GNU*****************************************************************
+File: bellman_ford.c
+
+Generated with Template by:
+Copyright (c) 2015 pgRouting developers
+Mail: project@pgrouting.org
+
+Function's developer:
+Copyright (c) 2017 Celia Virginia Vergara Castillo
```
Comment (cvvergara) [inline src/bellman_ford/src/bellman_ford.c]: I am not the function's developer

File: tools/template/create.sh [hunk 2]
```diff
@@ -29,7 +29,7 @@ fi
 
 
 #dont put pgr_ on MY_FUNCTION_NAME that is done automatically
-MY_FUNCTION_NAME="funnyDijkstra"
+MY_FUNCTION_NAME="funny_dijkstra"
```
Comment (cvvergara) [inline tools/template/create.sh]: Next time you want get the original file do a:
git checkout -- <name of file>
No harm done with this change

File: include/drivers/bellman_ford/bellman_ford_driver.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
+/*PGR-GNU*****************************************************************
+File: bellman_ford_driver.h
+
+Generated with Template by:
+Copyright (c) 2015 pgRouting developers
+Mail: project@pgrouting.org
+
+Function's developer:
+Copyright (c) 2015 Celia Virginia Vergara Castillo
```
Comment (cvvergara) [inline include/drivers/bellman_ford/bellman_ford_driver.h]: I am not the function's developer

Comment (codeSG) [reply include/drivers/bellman_ford/bellman_ford_driver.h]: will do the required change

**Comment by cvvergara** [comment]
About https://github.com/pgRouting/pgrouting/pull/1033/commits/d4cc29d427d843bab1898a7d029cd7ab85a838a8

There is already a run.sh file that you copy locally to your computer and adjust it to your needs, and never include it in the repo

make a backup of your simplified run.sh and do a `git rm run.sh` then move your copy back to run.sh and just use it.

For example:
if I do a git status I get a zillion of files for me to use while I develop but dont need to be on the repo
```
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        changes.txt
        check.txt
        code.txt
        install.txt
        ... (like 20 more)
        publish.sh
        result1.txt
        run.sh
        sql/scripts/extension-update-files.py
        sql/scripts/outfile.sql
        test.txt
        test/trailing.sh
        tools/developer/install_gcc.sh
        trailing.sh
        vrp.txt
```

**Comment by codeSG** [comment]
OKay ma'am. Got it.

**Pull request merged**